Lib/test/test_pathlib.py

```diff
@@ -3002,6 +3002,25 @@ def test_absolute(self):
             self.assertEqual(str(P('a', 'b', 'c').absolute()),
                              os.path.join(share, 'a', 'b', 'c'))
 
+        drive = os.path.splitdrive(BASE)[0]
+        with os_helper.change_cwd(BASE):
+            # Relative path with root
+            self.assertEqual(str(P('\\').absolute()), drive + '\\')
+            self.assertEqual(str(P('\\foo').absolute()), drive + '\\foo')
+
+            # Relative path on current drive
+            self.assertEqual(str(P(drive).absolute()), BASE)
+            self.assertEqual(str(P(drive + 'foo').absolute()), os.path.join(BASE, 'foo'))
+
+        with os_helper.subst_drive(BASE) as other_drive:
+            other_cwd = f'{other_drive}\\dirA'
+            with os_helper.change_cwd(other_cwd):
```

Comment:
`change_cwd()` can't be used here. For some reason, it first calls `realpath()`. POSIX requires `getcwd()` ([spec](https://pubs.opengroup.org/onlinepubs/9699919799/functions/getcwd.html)) to return a real path (i.e. no symlinks). Maybe `realpath()` is called for cross-platform consistency. Windows doesn't care whether `GetCurrentDirectoryW()` or `GetFullPathNameW()` returns a real or final path, and our implementation of `os.getcwd()` does nothing to comply with POSIX in this regard.

On Windows, `realpath()` is based on `GetFinalPathNameByHandleW()`. Say the current NT logon session has a substitute drive `\\??\\Z:` that resolves to `\\??\\C:\\Temp`, which in turn resolves to "\\Device\\HarddiskVolume2\\Temp". We start by opening a handle for this path, and we pass it to `GetFinalPathNameByHandleW()`. It queries the filesystem for the final normalized path (e.g. resolve short names as long names), which is "\\Temp". It queries the object manager for the device object name, which is "\\Device\\HarddiskVolume2". It queries the mount-point manager to get the canonical DOS name of the volume device, which is "\\??\\C:"[^1]. It switches to using the WinAPI "\\\\?\\" prefix, and it joins and returns the final path "\\\\?\\C:\\Temp".

[^1]: The canonical DOS name of a volume may be a mount point (i.e. a junction) in a filesystem, such as "\\??\\C:\\Mount\\SpamVolume". One can also request the GUID name of the volume device, such as "\\??\\Volume{12345678-1234-1234-1234-123456781234}".